### PR TITLE
[CET-7367] typeFilter is now optional for CompoundFilter

### PR DESCRIPTION
**Pull request by PavelZurek**

### CODE REVIEW

**Review by FoodProduct — COMMENTED** [review]
What is the background of this change? An optional + nullable field is typically a code smell.

**Comment by PavelZurek** [comment]
@FoodProduct This is fix for the failed build in the backstage plugin PR.
https://github.com/cortexapps/backstage-plugin/actions/runs/8570805375/job/23489571799?pr=156
This fails because the function is called on this object to convert all `null` to `undefined`, so I think we need both here

File: src/extensionApi.ts [hunk 1]
```diff
@@ -278,7 +278,7 @@ export interface CompoundFilter {
   cqlFilter?: GenericCqlFilter;
   entityGroupFilter?: EntityGroupFilter;
   type: FilterType.COMPOUND_FILTER;
-  typeFilter: CatalogPageTypeFilter | null;
+  typeFilter?: CatalogPageTypeFilter | null;
```
Comment (FoodProduct) [inline src/extensionApi.ts:281]: Alright so then that implies that the correct typing is 
```suggestion
  typeFilter?: CatalogPageTypeFilter;
```
right?

Comment (PavelZurek) [reply src/extensionApi.ts]: Oh yes, sorry. I tested this and it works good - fixed.

**Review by FoodProduct — APPROVED** [review]

**Pull request merged**